common/scala/src/main/scala/whisk/core/mesos/MesosContainerFactory.scala

```diff
@@ -126,9 +139,28 @@ class MesosContainerFactory(config: WhiskConfig,
       //strip any "--" prefixes on parameters (should make this consistent everywhere else)
       parameters
         .map({ case (k, v) => if (k.startsWith("--")) (k.replaceFirst("--", ""), v) else (k, v) })
-        ++ containerArgs.extraArgs)
+        ++ containerArgs.extraArgs,
+      parseConstraints(constraintStrings))
   }
 
+  /**
+   * Validate that constraint strings are well formed, and ignore constraints with unknown operators
+   * @param constraintStrings
+   * @param logging
+   * @return
+   */
+  def parseConstraints(constraintStrings: Seq[String])(implicit logging: Logging): Seq[Constraint] =
+    constraintStrings.flatMap(cs => {
+      val parts = cs.split(mesosConfig.constraintDelimiter)
+      require(parts.length == 3, "constraint must be in the form <attribute><delimiter><operator><delimiter><value>")
+      Seq(LIKE, UNLIKE).find(_.toString == parts(1)) match {
+        case Some(o) => Some(Constraint(parts(0), o, parts(2)))
+        case _ =>
+          logging.warn(this, s"ignoring unsupported constraint operator ${parts(1)}")
+          None
+      }
+    })
```

Comment:
Just to leave the note, this could make use of Scala's pattern matching, as in:

```scala
cs.split(mesosConfig.constraintDelimiter).toList match {
  case a :: LIKE.toString :: b   => Some(Constraint(a, LIKE, b)
  case a :: UNLIKE.toString :: b => Some(Constraint(a, UNLIKE, b))
  case a ::o :: b                =>
    logging.warn(this, s"ignoring unsupported constraint operator $o")
    None
  case failure =>
    logging.warn(this, s"couldn't parse constraint $failure")
    None
}
```

This wouldn't throw an exception in any case and is typechecked throughout (`parts(i)` will throw on an out-of-bounds exception)